[Tim-Siu] iP

Tim-Siu opened the pull request:
# Heading: Duke-clone 🚀

> **Blockquote**: “Your mind is for having ideas, not holding them.” - David Allen ([source](https://www.example.com))

**Bold Statement**: DukePro frees your mind of having to remember things you need to do. 

_Bullet List:_
- **text-based**
- *italic: easy to learn*
- ~~strikethrough: FAST~~ SUPER FAST to use

_Numbered List:_
1. [Download it from here](https://github.com/tim-siu/ip)
2. Double-click it
3. `Inline code: add your tasks`
4. Let it manage your tasks for you 😉
5. And it's ***FREE***!

## Task List:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you're a `Java` programmer, you can use it to practice Java too. Here's the main method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Tim-Siu committed:
Updated .gitignore

Tim-Siu committed:
Level 0. Rename, Greet, Exit

Tim-Siu committed:
Level 1. Echo

Tim-Siu committed:
Level 2.Add, List

Tim-Siu committed:
Level 3.Mark as Done

Tim-Siu committed:
Interim update towards Level-4

Tim-Siu committed:
Level 4. ToDos, Events, Deadlines. Code optimized for readability. Introduced switch statement.

Tim-Siu committed:
A-TextUiTesting

Tim-Siu committed:
Level 5. Handle Errors. Still requires fine-tuning.

Tim-Siu committed:
Level-6 Delete. test also updated.

Tim-Siu committed:
A-Enums. Not fully done.

Tim-Siu committed:
Updated syntax to meet requirement

Tim-Siu committed:
Increment to Level-7 with added support for storage

Tim-Siu committed:
Merge branch 'branch-Level-7'
Merge level-7 branch to master

Tim-Siu committed:
Increment to Level-8 with advanced Time features added

Tim-Siu committed:
Merge branch 'branch-Level-8'
Merge the level-8 branch onto master

Tim-Siu committed:
Refactor the code without full functionality

Tim-Siu committed:
Increment to more OOP

Tim-Siu committed:
Increment to using packages

Tim-Siu committed:
Create the storage file when it doesn't exit

Tim-Siu committed:
Adjust the exception message

Tim-Siu committed:
Merge branch 'add-gradle-support'
Merge the add-gradle-support

Tim-Siu committed:
Increment to use gradle as the build tool

Tim-Siu committed:
Increment to A-JUnit

NoCommandDetailException is renamed to CommandDetailException

Tim-Siu committed:
Enhanced exception handling

Tim-Siu committed:
Added support for Javadoc

Tim-Siu committed:
Merge branch 'branch-A-JavaDoc'
Merge the javadoc branch

Tim-Siu committed:
Comply with coding standards

Tim-Siu committed:
Merge branch 'branch-A-CodingStandard'
Merge the coding standards update

Tim-Siu committed:
Increment to level 9 with further correction of error handling

Tim-Siu committed:
Merge branch 'branch-Level-9'
Merge with level 9 branch

Tim-Siu committed:
Configure the project to use check-style

Tim-Siu committed:
Follow the tutorial to add the interface. Code heavily bollowed from the tutorial

Tim-Siu committed:
Basic GUI created with dummy response

Tim-Siu committed:
Fully-functional GUI and better java doc comments added.

Tim-Siu committed:
Merge branch 'branch-Level-10'
Merge the branch-Level-10

Tim-Siu committed:
Attempt to add varargs into the code, but it is not applicable.

timleow commented on src/main/java/duke/commands/ClearCommand.java:
```diff
@@ -0,0 +1,59 @@
+     * @throws CommandDetailException If there is an error in the details of the command.
+     */
+    @Override
+    public String executeGui(TaskList tasks, Ui ui, Storage storage) throws StorageException, CommandDetailException {
```
Is this line a little too long?

timleow commented on src/main/java/duke/time/Time.java:
```diff
@@ -0,0 +1,54 @@
+import duke.exceptions.TimeParsingException;
+
+/**
+ * Provides utility methods for parsing and formatting date-time representations in the Duke application.
```
I like the neat header comments and javadocs! 💯

timleow commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,121 @@
+        String command = splitInput[0].toLowerCase();
+        try {
+            switch (command) {
+            case "bye": {
```
Great job following the right indentation for switch statements!

timleow commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,121 @@
+                if (task != null) {
+                    return new AddCommand(task);
+                } else {
+                    throw new CommandNotRecognizedException("I'm sorry, but I don't know what " + input + " means :-(");
```
This line might be a little too long too. Perhaps you could separate the string concatenation on different lines?

```java 
{
  throw new CommandNotRecognizedException("I'm sorry, but I don't know what " 
      + input 
          + " means :-(");
}
```

timleow reviewed (commented):
It was pleasurable to read your code as the vast majority of it followed the coding standard guidelines. Just a number of lines exceeding the recommended length, but other than that LGTM! 💯

Tim-Siu committed:
Increment to use Assert

There are many invariants in the program, especially in code related to parsing.

Adding assert lets us know if the code is working as we expect.

Let's,
* add some assert in both Parser.java to make sure the length of the String is as expected. I am not sure if I should add more since the catch block seems to be working well
* add assert in Time to make sure the time received is not empty

Tim-Siu committed:
Refactor code to follow code quality standard

There was some code that was too long and display arrow-head pattern.

Refactoring it improves readability.

As a step towards better readability, Iet's refactor the load method into three methods.

Tim-Siu committed:
Merge pull request #1 from Tim-Siu/branch-A-Assertions

Increment to use Assert

Tim-Siu committed:
Merge branch 'master' into branch-A-CodeQuality
Merge master to the feature branch

Tim-Siu committed:
Merge branch 'branch-A-CodeQuality'
Merge the feature branch to master

Tim-Siu committed:
Implement the sort functionality

Tim-Siu committed:
Merge branch 'branch-B-Sort'
Merge the extension branch B-Sort

Tim-Siu committed:
Support CI

Tim-Siu committed:
Update the CI configuratioin

Tim-Siu committed:
Update the CI configuratioin

Tim-Siu committed:
improved Gui

Tim-Siu committed:
Merge branch 'branch-A-BetterGui'
Merge the BetterGui branch.

Tim-Siu committed:
Added personalities

Tim-Siu committed:
Merge branch 'branch-A-Personality'
Merge the personality branch

Tim-Siu committed:
Added the welcome message

Tim-Siu committed:
Update the readme

Tim-Siu committed:
Tweek the Readme for improved readability

Tim-Siu committed:
Improve code style

Tim-Siu committed:
Solve the label collapse issue

Tim-Siu committed:
Added the product screenshot

Tim-Siu committed:
Resolve the cross-platform issue
